worlds/hk_rework/state_mixin.py

```diff
@@ -848,55 +913,55 @@ def TryMatch(cls, term: str):
     # def GetTerms(cls):
     #     return (term for term in ("VessleFragments",))
 
-    def has_item(self, item_state, player):
+    def has_item(self, item_state: CollectionState, player: int) -> bool:
         if self.void:
             count = 3
         else:
             count = 2
         return item_state._hk_processed_item_cache[player]["WHITEFRAGMENT"] >= count
 
-    # def _ModifyState(self, state_blob, item_state, player):
+    # def _ModifyState(self, state_blob: Counter, item_state: CollectionState, player: int):
     #     # TODO figure this out
     #     # TODO actually
     #     print(self.void)
     #     return super()._ModifyState(state_blob, item_state, player)
 
 
 class FlowerProviderVariable(RCStateVariable):
-    prefix = "$FLOWERGET"
+    prefix: str = "$FLOWERGET"
 
     # @classmethod
     # def GetTerms(cls):
     #     return (term for term in ("VessleFragments",))
 
-    def _ModifyState(self, state_blob, item_state, player):
+    def _ModifyState(self, state_blob: Counter, item_state: CollectionState, player: int) -> tuple[bool, Counter]:
         state_blob["NOFLOWER"] = False
         return True, state_blob
 
     @classmethod
-    def TryMatch(cls, term: str):
+    def TryMatch(cls, term: str) -> bool:
         return term == cls.prefix
 
-    def can_exclude(self, options):
+    def can_exclude(self, options: HKOptions) -> bool:
         return False
 
 
 class RegainSoulVariable(RCStateVariable):
-    prefix = "$REGAINSOUL"
+    prefix: str = "$REGAINSOUL"
     amount: int
 
-    def parse_term(self, amount):
+    def parse_term(self, amount: int | str | float) -> None:  # qwint what type is this supposed to be?
```

Comment:
parse_term always comes in with str from the json (through the constructor)
```suggestion
    def parse_term(self, amount: str) -> None:
```